packages/apollo/src/utils/lists.ts

```diff
@@ -86,7 +86,18 @@ export class ListHelper {
             return undefined
         }
 
-        return existing.slice(skip, skip + first)
+        const currentList = existing.slice(skip, skip + first).filter(Boolean)
+
+        const cache = options?.cache.extract()
+
+        const hasBrokenLinks = currentList.some((link) => {
```

Comment:
You need to add a big explanation to this code, as it is a bit uncliear from the get go